### ISSUE TITLE
More control over fuel consumption

### ISSUE THREAD

**Issue by exentio** [SlimeKnights/TinkersConstruct#4569]
**Is your idea related to a problem? Please describe.**
I think the new fuel consumption system is way too aggressive and punishing: a smeltery with the size of 5x5x5 internal, which I'd consider standard and not that big considering the maximum size is 14x14x64 internal, has a fuel consumption of 50mb every three seconds. It's way too high to be usable, especially considering that lava is not a renewable resource and that the consumption is not related to the number of items being smelted.

**Describe the solution you'd like**
More controls over the fuel consumption in the config files. An option to use the TiC 2 way to calculate fuel consumption would be cool too, but what's important is having some way to change the consumption in the first place.

**Describe alternatives you've considered**
Different fuel tiers? But let's not overcomplicate things, TiC 3 feels way more punishing than TiC 2 and I can't say I'm digging it.

**Versions:**
* Minecraft: 1.16.5
* Forge: 36.1.32
* Mantle: 1.6.109
* Tinkers Construct: 3.0.4.197 (I'm aware of the new version)

**Confirm below that this enhancement is not covered on the roadmap or "Whats New?"**
The enhancement is not covered on the roadmap, and it's a change to the current mechanics

**Comment by KnightMiner** [comment]
TiC 2 basically used no fuel, its not a fair comparison. The changes were because it was silly how little fuel it used. Anyways, to directly address some comments:

* A 5x5x5 smeltery does use a lot of fuel, it can also melt 125 items at once. I think for 125 items at a time, 50mb every 3 seconds is fair
* Lava may not be renewable, but there is a giant ocean of lava in the nether, plus a renewable fuel source through blazes
* More controls over the fuel control just means it will be inconsistent among modpacks, so a build that works in one pack does not work in another. Tinkers has never put its primary numbers like that in config, so I don't see much reason to start now.

Overall, I think its good that bigger smelteries are not always better. Means you actually have to think about how big you want your smeltery instead of just spamming more blocks on it. You also always have the option of multiple smelteries, if your goal was liquid storage. Try playing with this mechanic and figure out how to work with the limitation instead of just saying its bad because its nerfed

So tl;dr: blazing blood is renewable, TiC 2 fuel cost was basically non-existent, and tradeoffs are a good thing.

**Comment by exentio** [comment]
Pretty sure this deserved at least a bit of taking, instead of an instant close after putting words I never said in my mouth.

>TiC 2 basically used no fuel, its not a fair comparison  

I agree, which is why I made no direct comparison. I choose the word "punishing" for a good reason, I didn't just say "it's harder": it's nice to have a TiC that doesn't just hand you everything for free, but I'd also like it if it didn't make me go through hell without a good reason.  
Mine was just a proposal because why not, I imagined it to be tuneable to make it more balanced. After all, personalization is kinda the point of mods in the first place, and having options is nice.

> I think for 125 items at a time, 50mb every 3 seconds is fair  

I also agree with this. What I don't agree with is that the fuel consumption stays the same when I'm trying to smelt just a couple of blocks, this is NOT fair and it's not balanced in any way.

>More controls over the fuel control just means it will be inconsistent among modpacks  

Mods have been giving control over consumption since their beginnings, it's never been an issue. Configs are also optional, it's the modpack creator's duty to communicate every change to the configs.

>Tinkers has never put its primary numbers like that in config, so I don't see much reason to start now  

This is far from being a valid argument, the mod is going under a complete rewrite but adding configs is incoherent for some reason?

>instead of just saying its bad because its nerfed  

I never said it's bad, please don't put words in my mouth. I said it's excessive, which is very different: I welcome the change, but not the way it has been implemented.  
Nerfing means making the TiC smeltery, that had very low fuel consumption, actually start using fuel. 50mb of fuel independently from the number of blocks is NOT nerfing, it's completely unbalanced.  

The reasons why I said that I'm not digging the TiC 3 are not related to the smeltery, and in the end I just have to get used to them. However, what I said in this issue is different, I'd rather have the possibility to change things in the config and have an up-to-date mod, than have to wait for a fork that either makes it less painful or straight up just ports TiC 2 to the latest Minecraft versions

**Comment by KnightMiner** [comment]
> Pretty sure this deserved at least a bit of taking, instead of an instant close after putting words I never said in my mouth.

This is a bug tracker, the enhancement tag is mainly for quick things that need no debate. We have a discord server if you want to discuss things in depth.

> I also agree with this. What I don't agree with is that the fuel consumption stays the same when I'm trying to smelt just a couple of blocks, this is NOT fair and it's not balanced in any way.

Its perfectly fair, you made a giant smeltery, you pay giant fuel costs. Don't want to pay giant fuel costs? Make a smaller smeltery.

Charging fuel per tick and not per item has always been the way the smeltery worked, so if you want max efficiency fill it up completely. If you don't want to make a smaller smeltery to save fuel, smelt items in batches instead of one at a time. Be a little creative instead of asking for the block to magically be the most efficient. Not only is that unrealistic, its just poor gameplay in my opinion.

> Mods have been giving control over consumption since their beginnings, it's never been an issue. Configs are also optional, it's the modpack creator's duty to communicate every change to the configs.

Heh, you know how many issue reports I get about a feature added for one of my mods in a single modpack that was clearly communicated about from that modpack as being a feature added by that modpack? Players constantly assume that because something is true in one pack, its true in all packs. Especially when that number is an internal number that most players will not understand.

We have support for changing the length of time lava lasts, if for some reason lava is so precious, use a datapack or crafttweaker to make lava burn for longer. That is something players can easily understand, so that is the config that makes sense to use here.

> This is far from being a valid argument, the mod is going under a complete rewrite but adding configs is incoherent for some reason?

Adding configs for mechanics I don't think should be changed is the incoherent thing. I see no reason why one modpack would want fuel to last longer in large smelteries, all packs I have seen either have lava gen (so that number is effectively irrelevant) or mean fuel to be at least somewhat of a challenge (so would not want the fuel to be consumed slower). If you dislike the current rate, argue why the current rate is bad, don't suggest adding configs so you can just ignoring my design.

> 50mb of fuel independently from the number of blocks is NOT nerfing, it's completely unbalanced.

Its dependent on the number of blocks making up your smeltery. So to repeat myself again, *make smaller smelteries if you care about fuel so much*, or *smelt items in batches*.

Overall, your arguments sound like you have a solution you want to use and you keep arguing to use that solution, instead of considering other options. Just because a sledge hammer does not work well to pound in a nail does not mean its poorly designed, it means you are misusing it.

----

Anyways, if this change is so terrible, you are free to make your own mod. If this is not the reason you are "not digging" the mod, I don't get why you are discussing it so much. As I said, you are free to join our discord (which was linked on the new issue screen) to discuss things further. Also, overall I highly doubt anyone is going to port TiC 2 to 1.16 because the smeltery uses a little more fuel that you want, TiC code is quite complicated to undergo such a huge effort for such a small complaint.
